### PR TITLE
Add GitHub.com themes

### PR DESCRIPTION
**Pull request by simurai**
This adds the `GitHub.com Light` + `GitHub.com Dark` themes. It's the same as #102 (that got reverted), but still keeps the old themes.

- `GitHub Light` stays as is
- `GitHub Dark` stays as is
- `GitHub Light Default` ✨  new ✨ 
- `GitHub Dark Default` ✨  new ✨ 
- `GitHub Dark Dimmed` ✨  new ✨ 

GitHub.com Light | GitHub.com Dark | GitHub.com Dimmed
--- | --- | ---
Screen Shot 2021-03-24 at 12 09 35 | Screen Shot 2021-03-24 at 12 08 36 | Screen Shot 2021-03-24 at 12 07 58

## TODO:

- [x] Decide what names to use
- [ ] Wait for final approval

**Note**: Currently `GitHub Light` and `GitHub Light Default` are almost the same. A bit different contrast for text. It almost feels redundant having both. But who knows, that could change in the future. 🤷

### CODE REVIEW

**Review by dominicegginton — APPROVED** [review]

**Comment by Gameghostify** [comment]
@simurai Thank you for this PR, this feels like a really good solution

**Review by samuelramox — APPROVED** [review]

**Review by mishushakov — APPROVED** [review]

**Comment by davi-mbatista** [comment]
Awesome. Can't wait to have it merged!

**Comment by bogdansoare** [comment]
Anything stopping this PR from being merged?

**Comment by simurai** [comment]
> Anything stopping this PR from being merged?

I think we still have to decide if `GitHub.com Dark` is the right name moving forward. /cc @broccolini 

---

If you already wanna try the github.com theme, you can go back to a previous version by clicking on the dropdown next to "Uninstall":

Screen Shot 2021-02-10 at 8 55 03

Then choose "Install Another Version..." and pick `2.0.1`:

Screen Shot 2021-02-10 at 9 00 16

**Comment by duong-le** [comment]
How about this?
- GitHub.com Light, GitHub Light -> GitHub Light (Since they are almost the same)
- GitHub Dark -> GitHub Dark Classic
- GitHub.com Dark -> GitHub Dark

File: src/dotcom/theme.js
Comment (dominicegginton) [inline src/dotcom/theme.js:183]: Once merged the [color-modes](https://github.com/primer/primitives/tree/mkt/color-modes) branch in the [primer/primitives](https://github.com/primer/primitives) module will provide ansi colors for the terminal, we should add these to this theme
``` js
'terminal.ansiBlack': color.ansi.black,
'terminal.ansiRed': color.ansi.red,
'terminal.ansiGreen': color.ansi.green,
'terminal.ansiYellow': color.ansi.yellow,
'terminal.ansiBlue': color.ansi.blue,
'terminal.ansiMagenta': color.ansi.magenta,
'terminal.ansiCyan': color.ansi.cyan,
'terminal.ansiWhite': color.ansi.white,
'terminal.ansiBrightBlack': color.ansi.blackBright,
'terminal.ansiBrightRed': color.ansi.redBright,
'terminal.ansiBrightGreen': color.ansi.greenBright,
'terminal.ansiBrightYellow': color.ansi.yellowBright,
'terminal.ansiBrightBlue': color.ansi.blackBright,
'terminal.ansiBrightMagenta': color.ansi.magentaBright,
'terminal.ansiBrightCyan': color.ansi.cyanBright,
'terminal.ansiBrightWhite': color.ansi.whiteBright,
```

**Review by dominicegginton — CHANGES_REQUESTED** [review]

**Comment by dominicegginton** [comment]
@simurai, just noticed that GitHub will be adding, a [dimmed](https://github.com/github/roadmap/issues/175) and [deuteranomaly colorblind](https://github.com/github/roadmap/issues/176) theme to GitHub.com. We should add these to the vscode theme extension to keep consistent. This could clear up some of the confusion that was caused when dark mode was released for this extension. Let me know if you need any help on this 😄

**Comment by dominicegginton** [comment]
### Suggestion
With the addition of the new GitHub.com dimmed theme recently merged into this PR (#138), I believe we can simplify theme names and remove the original themes reducing confusion and simplifying user experience. My suggestion is:

**Remove**
- `GitHub Light`: Due to high similarity to **GitHub.com Light**
- `GitHub Dark`: Due to high similarity to **GitHub.com Dimmed**

As you can see it is clear that the new old themes are extremely similar to the new `.com` themes. These old themes have been forked as discussed in #114 and the new dimmed theme addresses the main concern of contrast that resulted in the rollback of the new dark theme in the first place. Due to this, I believe it is not worth supporting the old thems Moving forward. 
|GitHub Light|GitHub.com Light|
|----------------|------------------------|
|screenshot-Wed-17Mar21_13 15|screenshot-Wed-17Mar21_13 15_000|

|GitHub Dark|GitHub.com Dimmed|
|----------------|----------------------------|
|screenshot-Wed-17Mar21_13 17|screenshot-Wed-17Mar21_13 18|




**Rename**
- `GitHub.com Light`  :arrow_right:  `GitHub Light`
- `GitHub.com Dark`  :arrow_right:  `GitHub Dark`
- `GitHub.com Dark Dimmed`  :arrow_right:  `GitHub Dimmed`

If removing the original themes it makes sense to rename these removing the `.com`.

**Result**

This would leave us with these three themes that follow the new GitHub.com website:

|GitHub Light|GitHub Dark|GitHub Dimmed|
|-----------------|----------------|---------------------|
|screenshot-Wed-17Mar21_13 15_000|screenshot-Wed-17Mar21_13 28|screenshot-Wed-17Mar21_13 18|

If anyone has any comments or feedback for this suggestion I'm making please let me know. If this is something people and @simurai want I'm happy to open a PR for review.

**Comment by simurai** [comment]
**Update**: With https://github.com/primer/github-vscode-theme/pull/132/commits/ea0286529cb18141461f3e0a4feb65386b8a4c7e, the naming of the themes are:

- `GitHub Light` stays as is
- `GitHub Dark` stays as is
- `GitHub Light Default` ✨  new ✨ 
- `GitHub Dark Default` ✨  new ✨ 
- `GitHub Dark Dimmed` ✨  new ✨ 

I'm not a big fan of the word "Default" in the name, but it allows the old `Light/Dark` themes to stay and it matches how [github.com calls the themes](https://github.com/settings/appearance) (with switched order):

image

**Review by nonameolsson — APPROVED** [review]

**Comment by joshmanders** [comment]
@simurai with these changes can we see an update to the extension soon? I'd really like to turn auto updating for extensions back on. Every time update happens it updates this extension and I have to roll it back, so had to turn auto update off.

**Comment by stagefright5** [comment]
I didn't want to be "that" guy, but, come on, this PR has been open for more than 2 months, which is honestly quite ridiculous.
If naming is the concern, that can be decided and changed even after this gets released. Please merge this PR.
Thanks

**Comment by dominicegginton** [comment]
@stagefright5 I know a lot of people (including me) are excited about the new official themes that will be added with this PR. But I'm sure the GitHub primer team are busy and want to make sure they have a clear and consistent direction to their design system. Remember this is also the default theme for GitHub Codespaces so is important to the GitHub brand image that it's of high standard.

**Comment by simurai** [comment]
> If naming is the concern, that can be decided and changed even after this gets released.

[We tried to rename it before](https://github.com/primer/github-vscode-theme/discussions/114), but I think VS Code uses the name as the identifier, so we can't really change names later without causing disruptions.

> Remember this is also the default theme for GitHub Codespaces

Yep, this is not really a "stand-alone" theme and is part of GitHub's release cycles. So sometimes things can get postponed.

**Comment by joshmanders** [comment]
Regardless of how long it takes to release, thank you @simurai and everyone for your work. I've been a long time Dracula theme user, since Reno released it, everywhere a theme for it exists I used it... GitHub's new dark theme is the first time I ever saw a theme that made me want to switch.

**Comment by nero120** [comment]
This looks awesome guys, great work! Looks like the PR is awaiting "final approval" - just wondering who needs to do that as I'm mega impatient to use dark dimmed! Looks like all the dev work has been done right?

**Review by nonameolsson — APPROVED** [review]

**Comment by simurai** [comment]
> nonameolsson approved these changes

Ok, in that case, let's merge it. 🚢

**Pull request merged**
